src/main/java/htsjdk/variant/vcf/VCFHeader.java

```diff
@@ -579,11 +601,23 @@ public boolean hasFilterLine(final String id) {
     }
 
     /**
+     * First checks for a VCFHeaderLine with the given key.
+     * If not found, looks for the first ID header line with the given key as the field type.
+     *
      * @param key    the header key name
      * @return the meta data line, or null if there is none
      */
     public VCFHeaderLine getOtherHeaderLine(final String key) {
-        return mOtherMetaData.get(key);
+        if (mOtherMetaData.containsKey(key)) {
+            return mOtherMetaData.get(key);
+        } else if (mIdMetaData.containsKey(key)) {
+            // Get the first item in the linked hash map
+            Map<String, VCFSimpleHeaderLine> fieldMetaData = mIdMetaData.get(key);
+            if (fieldMetaData.keySet().size() > 0) {
+                return fieldMetaData.get(fieldMetaData.keySet().iterator().next());
+            }
+        }
+        return null;
```

Comment:
This is for backwards compatibility, where previously a `VCFSimpleHeaderLine` could be accessed by this method (only one per field type). It may be better to let this return `null` and force users to use `getIdHeaderLine()` to find `VCFSimpleHeaderLine`s. This would be more consistent with `getOtherHeaderLines`/`getIdHeaderLines`.

Reply:
While this method makes sense by itself, it introduces a new inconsistency with `getOtherHeaderLines()`, since it has a different idea of what "other" means (it returns lines that will never be returned by the existing method).